### ISSUE TITLE
Allow linkType on Reference*Field 

### ISSUE THREAD

**Issue by danilobuerger** [marmelab/admin-on-rest#740]
It would be nice if I can specify a linkType on ReferenceManyField or ReferenceArrayField so that when using a SingleFieldList the containing Field would be wrapped with Link.

I could then do something like:

```
<ReferenceManyField label="Comments by" reference="comments" target="post_id" linkType="show">
  <SingleFieldList>
    <ChipField source="author.name" />
  </SingleFieldList>
</ReferenceManyField>
```

Which would make the Chip link to the respective comment.

**Comment by danilobuerger** [comment]
@fzaninotto I would like to make a PR for this if it is something that is actually wanted?

A couple of implementation ideas:

a) Add `linkType` to `ReferenceManyField` and `ReferenceArrayField` with default `false` (to keep backward compatibility). Pass `linkType` to child. Implement rendering of `Link` in `SingleFieldList`.

b) Add `linkType` to `SingleFieldList` with default `false` (to keep backward compatibility). Implement rendering of `Link` in `SingleFieldList`.

c) Add a new component called `SingleLinkFieldList` (or similar). Add `linkType` to `SingleLinkFieldList` with default `edit` (same as `ReferenceField`). Implement rendering of `Link` in `SingleLinkFieldList`.

**Comment by fzaninotto** [comment]
I think a) makes more sense considering the existing implementation of `<ReferenceField>`

**Comment by ryankeener** [comment]
Any progress on this?

**Comment by fzaninotto** [comment]
Fixed by #1503
